azurerm/resource_arm_public_ip.go

```diff
@@ -60,6 +60,19 @@ func resourceArmPublicIp() *schema.Resource {
 				}, true),
 			},
 
+			"ip_version": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  string(network.IPv4),
+				ForceNew: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					string(network.IPv4),
+					string(network.IPv6),
+				}, true),
+				DiffSuppressFunc: ignoreCaseDiffSuppressFunc,
+				StateFunc:        ignoreCaseStateFunc,
```

Comment:
since this is a new field, can we make this case-sensitive? (e.g. `true` -> `false` in the validate func, and remove the DiffSuppressFunc/StateFunc)

Reply:
the issue's to do with casting the string value to an Enum - if the casing doesn't match, Go defaults to the last value in the Enum, meaning the wrong value could be selected; as such we need to ensure the correct casing is used here (e.g. `IPv4` rather than `ipv4`)

Reply:
I'm making the change, but I haven't seen this as an issue yet. wouldn't the only fear of this be in reading the data? And with the data object IPv6 is the last value, but the test still is able to read IPv4.  And there were many purposefully strangely cased examples in my tests. Where in particular could you see this coming from?

Reply:
indeed - it's possible for the casing to be different for example where resources are provisioned via an ARM Template/the CLI and then imported